frontend/app/routes/_gcweb-app.personal-information.home-address.edit.tsx

```diff
@@ -87,15 +88,20 @@ export async function action({ request }: ActionFunctionArgs) {
   const session = await sessionService.getSession(request);
   session.set('newHomeAddress', parsedDataResult.data);
 
-  // TODO the validateWSAddress(..) call below must be corrected to consume the correct response
-  // const wsAddressWebService = await getWSAddressService();
-  // const { address, city, country, postalCode, province } = parsedDataResult.data;
-  // const validateResultAddress = await wsAddressWebService.validateWSAddress({ address, city, country, postalCode: postalCode ?? '', province: province ?? '' });
-
-  // TODO the following line must be removed once the validateWSAddress(..) call above is corrected
-  const validateResultAddress = { statusCode: 'Valid' };
-
-  const redirectUrl = validateResultAddress.statusCode === 'Invalid' ? '/personal-information/home-address/address-accuracy' : '/personal-information/home-address/confirm';
+  const wsAddressService = await getWSAddressService();
+  const { address, city, country, postalCode, province } = parsedDataResult.data;
+  const correctedAddress = await wsAddressService.correctAddress({ address, city, country, postalCode: postalCode ?? '', province: province ?? '' });
+
+  let redirectUrl;
+  if (correctedAddress.status === 'Corrected') {
+    const { address, city, province, postalCode, country } = correctedAddress;
+    session.set('suggestedAddress', { address, city, province, postalCode, country });
+    redirectUrl = '/personal-information/home-address/suggested';
+  } else if (correctedAddress.status === 'NotCorrect') {
+    redirectUrl = '/personal-information/home-address/address-accuracy';
+  } else {
+    redirectUrl = '/personal-information/home-address/confirm';
+  }
```

Comment:
**Thinking out loud:** would something like this be easier to read?

``` tsx
const responseHeaders = ...;

if (correctedAddress.status === 'Corrected') {
  return ...;
}

if (correctedAddress.status === 'NotCorrect') {
  return ...;
}

return ...;
```

I just find the if/elseif/else pattern hard to read at a glance (especially in javascript, where horizontal whitespace seems to be discouraged by convention).

At least if you break it into individual ifs with a default fallback, you can add some whitespace around them to draw the eye to the exit hatch.

Reply:
I agree although I was struggling with a clean way of doing this because we have to commit the session after setting a session variable (in the `correctedAddress.status === 'Corrected'` block).

I just created another function to deal with the `redirectUrl` instead as seen [here](https://github.com/DTS-STN/canadian-dental-care-plan/compare/6d58543589c408a034c61f8a0209748698a0befc..3b8fa830bade1a4323acf74e1c574a7942d03d2c).